mono/io-layer/wthreads.c

```diff
@@ -358,7 +352,11 @@ GetThreadPriority (gpointer handle)
 				  (gpointer *)&thread_handle);
 				  
 	if (ok == FALSE) {
-		return (THREAD_PRIORITY_NORMAL);
+		pthread_getschedparam (pthread_self(), &policy, &param);
+		if ((policy == SCHED_FIFO) || (policy == SCHED_RR)) 
+			return (((MonoInternalThread *)handle)->priority);
```

Comment:
That won't work on windows: GetThreadPriority is a winapi function that wants a real HANDLE. If you have the managed thread object in hand, and you can tell the thread hasn't started, why can't you just return the stored priority right here (or better still in managed code)?